msal/tests/Test.MSAL.NET.Unit/PublicClientApplicationTests.cs

```diff
@@ -1087,5 +1087,87 @@ public void GetUserTest()
+                MockResult = new AuthorizationResult(AuthorizationStatus.UserCancel,
+                    TestConstants.AuthorityHomeTenant + "?error=user_canceled")
+            };
+
+            //add mock response for tenant endpoint discovery
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityHomeTenant)
+            });
+
+            MsalMockHelpers.ConfigureMockWebUI(ui);
+
+            try
+            {
+                AuthenticationResult result = await app.AcquireTokenAsync(TestConstants.Scope).ConfigureAwait(false);
+            }
+            catch (MsalClientException exc)
+            {
+                Assert.IsNotNull(exc);
+                Assert.AreEqual("authentication_canceled", exc.ErrorCode);
+                Assert.AreEqual("User canceled authentication", exc.Message);
+                return;
+            }
+            finally
+            {
+                Assert.IsTrue(HttpMessageHandlerFactory.IsMocksQueueEmpty, "All mocks should have been consumed");
+            }
+
+            Assert.Fail("Should not reach here. Exception was not thrown.");
+        }
+
+        [TestMethod]
+        [Description("Test for AcquireToken with access denied error. This error will occur if" +
+            "user cancels authentication with embedded webview")]
+        public async Task AcquireTokenWithAccessDeniedErrorTestAsync()
+        {
+            PublicClientApplication app = new PublicClientApplication(TestConstants.ClientId);
+
+            // Interactive call and authentication fails with access denied
+            MockWebUI ui = new MockWebUI()
+            {
+                MockResult = new AuthorizationResult(AuthorizationStatus.ProtocolError,
+                    TestConstants.AuthorityHomeTenant + "?error=access_denied")
+            };
+
+            //add mock response for tenant endpoint discovery
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityHomeTenant)
+            });
+
+            MsalMockHelpers.ConfigureMockWebUI(ui);
+
+            try
+            {
+                AuthenticationResult result = await app.AcquireTokenAsync(TestConstants.Scope).ConfigureAwait(false);
+            }
+            catch (MsalServiceException exc)
+            {
+                Assert.IsNotNull(exc);
+                Assert.AreEqual("access_denied", exc.ErrorCode);
+                return;
+            }
+            finally
+            {
+                Assert.IsTrue(HttpMessageHandlerFactory.IsMocksQueueEmpty, "All mocks should have been consumed");
+            }
+
+            Assert.Fail("Should not reach here. Exception was not thrown.");
```

Comment:
>Assert.Fail("Should not reach here. Exception was not thrown."); [](start = 12, length = 64)

I believe you can attribute the test case to expect an exception. Though this works as well and I honestly have no preference, not sure though which pattern has been used in the code already. (same above)

Reply:
I agree, but was keeping the same format as the other tests in this section.